File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/strategy/Operation.java
package designPatterns.behavioral.strategy;

public class Operation {
  static class Add implements Strategy {
    @Override
    public int doOperation(int num1, int num2) {
      return num1 + num2;
    }
  }

  static class Subtract implements Strategy {
    @Override
    public int doOperation(int num1, int num2) {
      return num1 - num2;
    }
  }

  static class Multiply implements Strategy {
    @Override
    public int doOperation(int num1, int num2) {
      return num1 * num2;
    }
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/strategy/Operation.kt
package behavioral.strategy

sealed class Operation {
  class Add : Strategy {
    override fun doOperation(num1: Int, num2: Int): Int = num1 + num2
  }

  class Subtract : Strategy {
    override fun doOperation(num1: Int, num2: Int): Int = num1 - num2
  }

  class Multiply : Strategy {
    override fun doOperation(num1: Int, num2: Int): Int = num1 * num2
  }
}
